lib/rubocop/rspec/top_level_group.rb

```diff
@@ -10,11 +10,21 @@ module TopLevelGroup
       def_node_matcher :example_or_shared_group?,
                        (ExampleGroups::ALL + SharedGroups::ALL).block_pattern
 
-      def on_block(node)
-        return unless respond_to?(:on_top_level_group)
-        return unless top_level_group?(node)
+      def on_new_investigation
+        top_level_groups.each do |node|
+          if respond_to?(:on_top_level_example_group)
+            example_group?(node, &method(:on_top_level_example_group))
+          end
 
-        on_top_level_group(node)
+          on_top_level_group(node) if respond_to?(:on_top_level_group)
+        end
+      end
+
+      def top_level_groups
+        return [] unless root_node
```

Comment:
To satisfy our "empty spec" smoke test. Is there a better way around this?

Reply:
return from the `on_new_investigation` if there is no ast?

Reply:
Good idea, thanks!

Reply:
Fixed.